### ISSUE TITLE
Release GIL

### ISSUE THREAD

**Issue by mrocklin** [conda/conda#10037]
I believe that there is a period during execution when Conda does not release the GIL.  

This doesn't affect the common case user and so isn't a priority, but I'm trying to solve multiple conda environments in parallel and would find this convenient.  As a workaround I'm happy to use processes, but if this is easy to do it might be worth doing as a matter of hygiene.  Feel free to close as low priority though if there is a lot more going on.

**Comment by jakirkham** [comment]
As Conda itself is pure Python, how would it release the GIL?

**Comment by mrocklin** [comment]
Systems running alongside conda during a solve sometimes hang.  My sense
was that maybe there was a step where Conda was linking out to some
external solver.  I could be mistaken though.  If so then this should be
closed.

**Comment by jakirkham** [comment]
Conda does use pycosat (a Python wrapper around PicoSAT). Though it seems to already be [releasing the GIL]( https://github.com/ContinuumIO/pycosat/blob/b38fd85b6f4dcc18efd6027e96e5785104f53bb0/pycosat.c#L219-L221 ). It's always possible there are other places the GIL may need to be released. However I think we would need to have a better idea as to where this hang occurs and whether that is solver related or not.

**Comment by mrocklin** [comment]
OK, happy to have this closed then
